src/overlays/actors/ovl_En_Kbt/z_en_kbt.c

```diff
@@ -31,36 +34,543 @@ const ActorInit En_Kbt_InitVars = {
     (ActorFunc)EnKbt_Draw,
 };
 
-#endif
+void EnKbt_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnKbt* this = THIS;
 
-extern UNK_TYPE D_06004274;
+    Actor_SetScale(&this->actor, 0.01f);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &object_kbt_Skel_00DEE8, &object_kbt_Anim_004274, this->jointTable,
+                       this->morphTable, OBJECT_KBT_LIMB_MAX);
+    this->unk_27C = 0;
+    this->actor.home.rot.z = 0;
+    this->unk_27E = 4;
+    this->unk_27F = 0;
+    this->unk_280 = 13;
+    this->unk_282 = 13;
+    this->unk_278 = func_80B3403C(globalCtx);
+    this->unk_284 = 0;
+    this->actor.textId = 0;
+    if (func_80B33E64(globalCtx)) {
+        func_80B33EF0(this, 6);
+        this->unk_282 = 11;
+        if (func_80B33E8C(globalCtx)) {
+            this->actor.textId = 0xC50;
+        } else {
+            this->actor.textId = 0xC4E;
+            this->unk_27C |= 1;
+        }
+        this->actionFunc = func_80B34314;
+    } else {
+        func_80B33EF0(this, 0);
+        this->unk_282 = 0;
+        this->actionFunc = func_80B34598;
+    }
+    this->actor.flags &= ~ACTOR_FLAG_1;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kbt/EnKbt_Init.s")
+void EnKbt_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kbt/EnKbt_Destroy.s")
+s32 func_80B33E64(GlobalContext* globalCtx) {
+    return gSaveContext.permanentSceneFlags[globalCtx->sceneNum].unk_14 & 1;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kbt/func_80B33E64.s")
+s32 func_80B33E8C(GlobalContext* globalCtx) {
+    if ((CURRENT_DAY == 3) ||
+        ((CURRENT_DAY == 2) && (gSaveContext.permanentSceneFlags[globalCtx->sceneNum].unk_14 & 2))) {
+        return true;
+    }
+    return false;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kbt/func_80B33E8C.s")
+void func_80B33EF0(EnKbt* this, s16 arg1) {
+    u8 sp38[] = {
+        ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_ONCE, ANIMMODE_LOOP, ANIMMODE_ONCE, ANIMMODE_LOOP, ANIMMODE_LOOP,
+        ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_ONCE, ANIMMODE_ONCE,
+    };
+    static AnimationHeader* sAnimations[] = {
+        &object_kbt_Anim_000670, &object_kbt_Anim_001674, &object_kbt_Anim_002084, &object_kbt_Anim_000FE8,
+        &object_kbt_Anim_001940, &object_kbt_Anim_00E7BC, &object_kbt_Anim_00F0C8, &object_kbt_Anim_002710,
+        &object_kbt_Anim_002DE0, &object_kbt_Anim_003414, &object_kbt_Anim_003D24, &object_kbt_Anim_001BF4,
+        &object_kbt_Anim_002084,
+    };
```

Comment:
I prefer in-function static to be above the nonstatics.

Reply:
I put it in data order, I didn't realise I could swap them.

Reply:
Oh, huh, didn't notice only one was... probably stuck with it as-is, then.